### ISSUE TITLE
bug: test exec breaks after updating from `4.11` to `4.12` - `node:crypto`

### ISSUE THREAD

**Issue by deleonio** [ionic-team/stencil#5358]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.12.x

### Current Behavior

We receive exceptions when executing the stencil tests.

### Expected Behavior

We receive **no** exception when executing the stencil tests.

### System Info

```shell
    System: node 20.9.0
    Platform: darwin (23.2.0)
   CPU Model: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 cpus)
    Compiler: /Users/***/Workspace/kolibri/lib/node_modules/.pnpm/@stencil+core@4.12.2/node_modules/@stencil/core/compiler/stencil.js
       Build: 1707755243
     Stencil: 4.12.2 🎯
  TypeScript: 5.3.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.27.0
```

### Steps to Reproduce

One-line command:

```shell
pnpm clean && pnpm i && pnpm i @stencil/core@4.11.0 && pnpm test && pnpm i @stencil/core@4.12.2 && pnpm test
```

1. Clone the repository: `git clone https://github.com/deleonio/kolibri-library.git`
2. Install the dependencies: `pnpm i`
3. Install Stencil to version 4.11: `pnpm i @stencil/core@4.11.0`
4. Run the tests: `pnpm test`
5. Update Stencil to version 4.12: `pnpm i @stencil/core@4.12.2`
6. Run the tests again: `pnpm test`

### Code Reproduction URL

`git clone https://github.com/deleonio/kolibri-library.git`

**Comment by deleonio** [comment]
Hi @rwaskiewicz,

I have provided a sampler repository to reproduce the issue here: https://github.com/deleonio/kolibri-library

**Comment by deleonio** [comment]
Here is a minimal sample for reproduction: https://github.com/deleonio/kolibri-library

**Comment by rwaskiewicz** [comment]
@deleonio 
Thanks! Is that reproduction public?  GitHub returns a 404 when I try to access it

**Comment by deleonio** [comment]
The repo is now public, sorry.

**Comment by alicewriteswrongs** [comment]
Hey @deleonio, I was just able to take a look at this and found a fix which seems to fix things in your repro case locally for me. I just published a [dev build containing the fix](https://www.npmjs.com/package/@stencil/core/v/4.12.2-dev.1708019928.b3d5775), if you wouldn't mind trying it out to confirm it fixes the issue!

You can do that like this:

```sh
npm i @stencil/core@4.12.2-dev.1708019928.b3d5775
```

**Comment by deleonio** [comment]
@alicewriteswrongs, thank you very much!

It works in the reproduction Repo and also in our large monorepo [Public UI / KoliBri](https://github.com/deleonio/kolibri) for the German gov.

If there is anything I can do for Stencil, please let me know. ❤️

**Comment by deleonio** [comment]
@alicewriteswrongs, thank you very much!

It works in the [reproduction Repo](https://github.com/deleonio/kolibri-library) and also in our large monorepo [Public UI / KoliBri](https://github.com/public-ui/kolibri) for the German gov.

If there is anything I can do for Stencil, please let me know. ❤️

**Comment by alicewriteswrongs** [comment]
Awesome, thanks for giving that a try! I'll get that merged and we'll ping here when it's released

**Comment by christian-bromann** [comment]
A fix for this has been published in [v4.12.3](https://github.com/ionic-team/stencil/releases/tag/v4.12.3).
